tutorials/elixir-phoenix-on-kubernetes-google-container-engine/index.md

```diff
@@ -0,0 +1,444 @@
+    following content into it. Alternately, you may
+    [download](https://github.com/GoogleCloudPlatform/community/tree/master/tutorials/elixir-phoenix-on-kubernetes-google-container-engine/Dockerfile)
+    a sample annotated Dockerfile to study and customize.
+
+        FROM elixir:alpine
+        ARG APP_NAME=hello
+        ARG PHOENIX_SUBDIR=.
+        ENV MIX_ENV=prod REPLACE_OS_VARS=true TERM=xterm
+        WORKDIR /opt/app
+        RUN apk update \
+            && apk --no-cache --update add nodejs nodejs-npm \
+            && mix local.rebar --force \
+            && mix local.hex --force
+        COPY . .
+        RUN mix do deps.get, deps.compile, compile
+        RUN cd ${PHOENIX_SUBDIR}/assets \
+            && npm install \
+            && ./node_modules/brunch/bin/brunch build -p \
+            && cd .. \
+            && mix phx.digest
+        RUN mix release --env=prod --verbose \
+            && mv _build/prod/rel/${APP_NAME} /opt/release \
+            && mv /opt/release/bin/${APP_NAME} /opt/release/bin/start_server
+        FROM alpine:latest
+        RUN apk update && apk --no-cache --update add bash openssl-dev
+        ENV PORT=8080 MIX_ENV=prod REPLACE_OS_VARS=true
+        WORKDIR /opt/app
+        EXPOSE ${PORT}
+        COPY --from=0 /opt/release .
+        CMD ["/opt/app/bin/start_server", "foreground"]
+
+    **Note:** If your app is named something other than `hello`, you will need
+    to modify the `APP_NAME` argument in the Dockerfile. If your app is an
+    umbrella app, you will also need to modify the `PHOENIX_SUBDIR` to contain
+    the path to the Phoenix application subdirectory, e.g. `apps/hello_web`.
+
+2.  Create a file called `.dockerignore` in your `hello` directory. Copy the
+    following content into it. Alternately, you may
+    [download](https://github.com/GoogleCloudPlatform/community/tree/master/tutorials/elixir-phoenix-on-kubernetes-google-container-engine/.dockerignore)
+    a sample annotated file to study and customize.
+
+        _build
+        deps
+        test
+        assets/node_modules
+
+    **Note:** if your app is an umbrella app, you may need to adjust the paths
+    to include the build, deps, and node_modules directories of the constituent
+    apps. In general, you want Docker to ignore artifacts that come from your
+    development environment, so it can perform clean builds.
+
+### Testing the Dockerfile
+
+Now build the image locally and test running your application from the image:
+
+    docker build --no-cache -t hello .
+    docker run -it --rm -p 8080:8080 hello
+
+The period at the end of the `docker build` command is required. It denotes the
+root directory of the application you are building.
+
+Visit [http://localhost:8080](http://localhost:8080) to see the Phoenix
+welcome screen running locally from your Docker image.
+
+## Deploy your application
+
+Now you're ready to deploy your application to Container Engine!
+
+### Building in the cloud
```

Comment:
While this is cool to show, I think it's sort of a distraction, no?

Reply:
How so? We need to get the image into gcr regardless. Either we run a single cloudbuild command to build and upload the image in a single go, or we build the dockerfile locally and then run a separate gcloud command to push the image.